articles/aprg-workflow.md

```diff
@@ -0,0 +1,30 @@
+
+Architecture is often expressed in diagrams. On the one hand they can give an overview of the system while on the other hand you can dig as deep as required into the details of a system without writing dozens of documents no one ever reads. Many current tools use already an XML or JSON format which could be checked into VCS. But anyone who ever merged conflicts in an XML document, knows that this is not the best you can do. Additionally XML or JSON format do not provide any (or at least less) semantic which could be executed. And often multiple diagrams show the different views on the same model. Every change of the model requires a change in all views. 
+
+Using architecture as code benefits here in all cases. Merging the code is much more intuitive, as the language can be the same as the source code itself. Also changes in the model are type safe (depending on the language you use eventually at execution time), so all views will be taken into account. 
+
+And finally code can be executed and tested. Architectural requirements which are expressed by code can be verified. The code can be processed by a CI/CD pipeline. After running the tests, the architecture can be deployed in a dedicated environment. 
+
+## Decisions, Scenarios and Tests
+
+A further step to bring the complete architecture model and the code together, is storing your architectural decision records (ADR) with your code. ADRs are short text records in a defined format. This helps to keep the decision simple but informative. See [ADR] for details about it.
+
+ADRs are made to fulfill quality attribute requirements for a system. When we now see our ADRs as the user stories of the architecture then the quality attribute requirements are acceptance criteria for these stories. 
+
+We could write tests against these requirements similar to the BDD approach für functional acceptance testing [todo BDD link]. As described in [EVOL] writing these tests can be a challenge, since there will not be the one recipe to rule them all. One could be an execution of a performance test in a special environment set up by the architecture as code scripts to test performance requirements. Another can be automated penetration tests for security or resilience tests following the principles of chaos engineering. 
+
+Important is that the architecture of the test system is setup using architecture code. Only then it is possible to automatically verify that the system to be delivered continuously still fits to all requirements. Since the effort for testing as usual must be balanced against the actual value it creates, it is advisable to follow a risk-driven approach to create ADRs and tests only for the parts of the system where the potential risk indeed requires it [RISK]. 
```

Comment:
hm. Perhaps we missed to have a red line in the articles. "Architecture as code" is the main topic of both articles. First of all it was one article, spread into two because of too many lines. Could be that this breaks the red line.